### ISSUE TITLE
Dataflow SpringBoot app fails to run when bundled as a fat uber jar

### ISSUE THREAD

**Issue by damccorm** [apache/beam#20245]
I have been working on a service that programmatically builds Dataflow pipeline using SpringBoot. My service runs perfectly when the application is started from the (IntelliJ) IDE or when I run with `./gradlew run` but fails when I create the fat-uber-jar.

The issue seems to be related to https://issues.apache.org/jira/browse/BEAM-1325 and another issue https://github.com/GoogleCloudPlatform/DataflowJavaSDK/issues/538 (although that is about an IDE and this is more broadly about development and deployment).

To verify, I opened the built jar and I see the following entries in the `MANIFEST.MF` file:

```

Manifest-Version: 1.0
Main-Class: org.springframework.boot.loader.JarLauncher
Start-Class: com.example.dataExtractor.ApplicationKt
Spring-Boot-Version:
2.2.5.RELEASE
Spring-Boot-Classes: BOOT-INF/classes/
Spring-Boot-Lib: BOOT-INF/lib/

```

This seems to be a pretty standard way of building SpringBoot apps and would appreciate some guidance in this direction.

I originally posted this as an issue to [GoogleCloudPlatform/DataflowJavaSDK](https://github.com/GoogleCloudPlatform/DataflowJavaSDK/issues/649) and they redirected me here.

Meanwhile, I have unblocked myself with a workaround which has its own drawbacks. 
I wrote an [article](https://suspendfun.com/2020/Dataflow-Springboot-app-fails-to-run-when-dockerised/) with more context if it helps.

Imported from Jira [BEAM-9669](https://issues.apache.org/jira/browse/BEAM-9669). Original Jira may contain additional context.
Reported by: raj-saxena.
